test/v3/services/zip_file_generator_test.rb

```diff
@@ -0,0 +1,57 @@
+require './lib/xapi/problem'
+require './test/v3_helper'
+
+class ZipFileGeneratorTest < Minitest::Test
+  def problem_attributes_recursive_files
+    {
+      track_id: 'fake',
+      language: 'Who Cares',
+      slug: 'one',
+      path: './test/fixtures',
+    }
+  end
+
+  def problem_attributes_simple_folder
+    {
+      track_id: 'animal',
+      language: 'Who Cares',
+      slug: 'dog',
+      path: './test/fixtures',
+    }
+  end
+
+  def get_files_name(file)
+    names = []
+    Zip::InputStream.open(file) do |io|
+      while (entry = io.get_next_entry)
+        names << entry.name
+      end
+    end
+    names
```

Comment:
Ah, this is the names of the files inside the zip.

Let's change the signature to `file_names(zip_file)`